examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -86,97 +95,150 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
         }
 
         const doTheMigration = async () => {
```

Comment:
Maybe add a comment either here or on ensureMigrating (or both) explaining the contract/calling patterns.  E.g. "we assume ensureMigrating will only be called once and resolve only after the migration is complete" vs. "ensureMigrating is called to start the migration, but also can be called if the migration is interrupted and needs to be reattempted (in which case it will have this state...)".  Bonus if these calling requirements can be validated/enforced with asserts.